src/tss2-esys/esys_tr.c

```diff
@@ -83,6 +83,7 @@ Esys_TR_Deserialize(ESYS_CONTEXT * esys_context,
     size_t offset = 0;
 
     _ESYS_ASSERT_NON_NULL(esys_context);
+    *esys_handle = esys_context->esys_handle_cnt++;
```

Comment:
This looks to be more than just a counting issue as the commit message states. Without this change it looks like the `esys_handle` returned to the caller was never set? It's hard to say since I'm not sure what happens to the `esys_handle` when it's passed to `esys_CreateResourceObject`. The comment block for this function however states that the `esys_handle` parameter is a reference to the `ESYS_TR` object being serialized which implies that it already has an `ESYS_TR` allocated? I don't believe this is actually true but if it is then we're overwriting a valud `ESYS_TR` handle. This does however look like a typo in the documentation that's furthering my confusion.

Reply:
You are correct. The commit message was wrong. Updated.